### PR TITLE
Move defaults to globalSettings (fix #296)

### LINKED ISSUE #296: `EclipseKeys.withSource in ThisBuild` is ignored (4.0.0 regression from 3.0.0)
I have a multi-project build, and set 
```
EclipseKeys.withSource in ThisBuild := true
```
in [the main build.sbt](https://github.com/inc-lc/ilc-scala/blob/master/build.sbt#L105). If I switch from 2.5.0 to 4.0.0, that stops working, because of the `in ThisBuild` part — somehow I need to set that per project. 3.0.0 is safe. Has that got anything to do with AutoPlugin?

### PR DESCRIPTION
**Pull request by Blaisorblade**
[SBT docs](http://www.scala-sbt.org/0.13/docs/Plugins.html#projectSettings+and+buildSettings) recommend:
“Use globalSettings to define the default value of a setting.”

`sbteclipse` doesn't do that, causing #296 and other variants. This PR fixes that bug. However, I'm not 100% sure that I'm moving all settings that should be moved.

This PR is on top of #298.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ import com.typesafe.sbt.SbtGit._
 
 object Build extends Build {
 
-  val baseVersion = "4.0.0"
+  val baseVersion = "4.0.1"
```
Comment (jsuereth) [inline project/Build.scala]: We acually need to bump this by major version if we remove thigns (break binary compatibility)

Comment (Blaisorblade) [inline project/Build.scala]: All removals are from #298, hence only removing dead code — sorry for the PR on top, can resubmit. But I'm still adding `globalSettings`, which arguably* breaks forward compatibility. So should it be 4.0.1, 4.1.0 or 5.0.0?

*Unlike for Scala, I never considered compatibility for plugins, and this isn't mentioned in http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html.

**Comment by jsuereth** [comment]
Nice changes!  LGTM, but this will require a major version bump, due to comaptibility changes.

Great work!

**Comment by Blaisorblade** [comment]
> major version bump.

So 5.0.0 it is. New PR coming.

**Pull request closed without merging**